### PR TITLE
updated nghttp2 component (IEC-132)

### PR DESCRIPTION
**Pull request by hrushikesh430**
Updated the nghttp2 submodule from v1.58.0 or v1.62.1. 

Testing -
   Tested this component using the example which is in sh2lib component. Checked whether this example runs before and           after upgrading the nghttp2

Binary size (http2_request.bin) -
        Before - 1015120 Bytes
        After - 1015536 Bytes
        
Library Size (libnghttp.a) -
       Before - 82712 Bytes
       After - 83087 Bytes

Link to changelog -
https://github.com/nghttp2/nghttp2/releases/tag/v1.62.1

### CODE REVIEW

File: coap/sbom_libcoap.yml [hunk 1]
```diff
@@ -1,9 +1,7 @@
 name: libcoap
-version: 4.3.4
+version: 4.3.5-rc2
```
Comment (AdityaHPatwardhan) [inline coap/sbom_libcoap.yml:2]: This seems to be marked as release candidate, are we expecting any changes in `v4.3.5` ?

**Comment by AdityaHPatwardhan** [comment]
@hrushikesh430 can you add more information about the changelog or maybe the link to the changelog ?

File: coap/sbom_libcoap.yml (hunk 1 above)
Comment (AdityaHPatwardhan) [reply coap/sbom_libcoap.yml:2]: Also I dont see the submodule for libcoap updated here.

**Review by mahavirj — APPROVED** [review]

**Review by AdityaHPatwardhan — APPROVED** [review]

**Review by hmalpani — APPROVED** [review]
LGTM

**Pull request merged**